src/subscript/field_statistics/field_statistics.py

```diff
@@ -967,50 +1007,120 @@ def get_specifications(input_dict, ertbox_size, ert_config_path):
             f"Missing keyword:  {key} specifying a list of iteration numbers "
             " for ensembles from ERT ES-MDA"
         )
-    key = "use_zones"
-    zone_names_used = copy.copy(list(zone_code_names.values()))
-    if key in input_dict:
-        zone_names_input = input_dict[key]
-        if zone_names_input is not None and len(zone_names_input) > 0:
-            zone_names_used = zone_names_input
-    check_use_zones(zone_code_names, zone_names_used)
 
-    key = "zone_conformity"
-    if key in input_dict:
-        zone_conformity = input_dict[key]
-    else:
-        raise KeyError(f"Missing keyword:  {key} specifying conformity per zone.")
-    check_zone_conformity(zone_code_names, zone_names_used, zone_conformity)
+    if not ertbox_size:
+        # ertbox size does not exist, read it from this scripts config file instead
+        print("ERTBOX size is not defined, need to get it from the config file")
+        key = "ertbox_size"
+        if key in input_dict["geogrid_fields"]:
+            ertbox_size = input_dict["geogrid_fields"][key]
+        else:
+            raise KeyError(
+                "Missing keyword 'ertbox_size'."
+                "Is required if the ERTBOX.EGRID is not found in the "
+                "configuration directory of the FMU project under: "
+                f" {ert_config_path / Path('../../rms/output/aps/ERTBOX.EGRID')}"
+            )
 
+    # Optional keywords
     key = "use_population_stdev"
     use_population_stdev = False
     if key in input_dict:
         use_population_stdev = input_dict[key]
 
+    use_geogrid_fields = False
+    zone_names_used = None
+    zone_conformity = None
+    zone_code_names = None
     param_name_dict = None
-    key = "continuous_property_param_per_zone"
+    disc_param_name_dict = None
+    key = "geogrid_fields"
     if key in input_dict:
-        param_name_dict = input_dict[key]
-    check_param_name_dict(zone_code_names, param_name_dict)
+        use_geogrid_fields = True
+        geogrid_fields_dict = input_dict[key]
```

Comment:
I would argue that the code will be more clear if you just use the key value directly i.e. `geogrid_fields` instead of putting it into `key` variable.

Reply:
I can fix this

Reply:
The main reason why I usually assign a key string to a variable is that it is used multiple times. So if the key text string is to be changed for some reason, I can do it only in one line. In the case here it is referred two times only so not so important to assign it to a variable.

Reply:
I keep it as it is where it is referred to 3 times or more.